src/tests/suite/daos_test.h

```diff
@@ -439,7 +439,17 @@ test_rmdir(const char *path, bool force)
 		if (!strcmp(ent->d_name, ".") || !strcmp(ent->d_name, ".."))
 			continue;   /* skips the dots */
 
-		sprintf(fullpath, "%s/%s", path, ent->d_name);
+		rc = snprintf(fullpath, PATH_MAX, "%s/%s", path, ent->d_name);
```

Comment:
It's a little less efficient but we may consider just using D_ASPRINTF which allocates sufficient space.   Essentially, in the loop you'd have

D_ASPRINTF and D_FREE with one additional D_FREE for escape cases after the out label.

Reply:
great, seems like D_ASPRINTF does what we should do here.